src/main/java/edu/wpi/grip/core/operations/composite/FilterContoursOperation.java

```diff
@@ -0,0 +1,110 @@
+            SocketHint.View.SPINNER);
+
+    @Override
+    public String getName() {
+        return "Filter Contours";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Find contours matching certain criteria.";
+    }
+
+    @Override
+    public Optional<InputStream> getIcon() {
+        return Optional.of(getClass().getResourceAsStream("/edu/wpi/grip/ui/icons/find-contours.png"));
+    }
+
+    @Override
+    public InputSocket<?>[] createInputSockets(EventBus eventBus) {
+        return new InputSocket<?>[]{
+                new InputSocket<>(eventBus, contoursHint),
+                new InputSocket<>(eventBus, minAreaHint),
+                new InputSocket<>(eventBus, minPerimeterHint),
+                new InputSocket<>(eventBus, minWidthHint),
+                new InputSocket<>(eventBus, maxWidthHint),
+                new InputSocket<>(eventBus, minHeightHint),
+                new InputSocket<>(eventBus, maxHeightHint),
+        };
+    }
+
+    @Override
+    public OutputSocket<?>[] createOutputSockets(EventBus eventBus) {
+        return new OutputSocket<?>[]{new OutputSocket<>(eventBus, contoursHint)};
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void perform(InputSocket<?>[] inputs, OutputSocket<?>[] outputs) {
+        final InputSocket<Contours> inputSocket = (InputSocket<Contours>) inputs[0];
+        final double minArea = ((Number) inputs[1].getValue()).doubleValue();
+        final double minPerimeter = ((Number) inputs[2].getValue()).doubleValue();
+        final double minWidth = ((Number) inputs[3].getValue()).doubleValue();
+        final double maxWidth = ((Number) inputs[4].getValue()).doubleValue();
+        final double minHeight = ((Number) inputs[5].getValue()).doubleValue();
+        final double maxHeight = ((Number) inputs[6].getValue()).doubleValue();
+
+        final MatVector inputContours = inputSocket.getValue().getContours();
+        final MatVector outputContours = new MatVector(inputContours.size());
+
+        // Add contours from the input vector to the output vector only if they pass all of the criteria (minimum
+        // area, minimum perimeter, width, and height)
+        int filteredContourCount = 0;
+        for (int i = 0; i < inputContours.size(); i++) {
+            final Mat contour = inputContours.get(i);
+            final Rect bb = boundingRect(contour);
+
+            if (contourArea(contour) < minArea) continue;
+            if (arcLength(contour, true) < minPerimeter) continue;
+            if (bb.width() < minWidth || bb.width() > maxWidth) continue;
+            if (bb.height() < minHeight || bb.height() > maxHeight) continue;
+
+            outputContours.put(filteredContourCount++, contour);
+        }
+
+        outputContours.resize(filteredContourCount);
+
+        final OutputSocket<Contours> outputSocket = (OutputSocket<Contours>) outputs[0];
+        outputSocket.getValue().rows = inputSocket.getValue().rows();
```

Comment:
Modify variables without a setter.
You shouldn't even be doing setting here. This should be a constructor call.